kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java

```diff
@@ -365,6 +371,43 @@
       "Whether to enable REST Proxy V3 API. Default is true.";
   private static final boolean API_V3_ENABLE_DEFAULT = true;
 
+  public static final String RATE_LIMIT_ENABLE_CONFIG = "rate.limit.enable";
+  private static final String RATE_LIMIT_ENABLE_DOC =
+      "Whether to enable request rate-limiting. Default is false.";
+  private static final boolean RATE_LIMIT_ENABLE_DEFAULT = false;
+
+  public static final String RATE_LIMIT_BACKEND_CONFIG = "rate.limit.backend";
+  private static final String RATE_LIMIT_BACKEND_DOC =
+      "The rate-limiting backend to use. The options are 'guava' and 'resilience4j'. Default is "
+          + "'guava'.";
+  private static final String RATE_LIMIT_BACKEND_DEFAULT = "guava";
+
+  public static final String RATE_LIMIT_PERMITS_PER_SEC_CONFIG = "rate.limit.permits.per.sec";
+  private static final String RATE_LIMIT_PERMITS_PER_SEC_DOC =
+      "The maximum number of permits to emit per second. A permit is an unit of cost for a "
+          + "request. More expensive requests will consume more permits. The cost for each "
+          + "resource/method can configured via rate.limit.default.cost and rate.limit.costs. "
+          + "Default is 50.";
+  private static final Integer RATE_LIMIT_PERMITS_PER_SEC_DEFAULT = 50;
+
+  public static final String RATE_LIMIT_TIMEOUT_MS_CONFIG = "rate.limit.timeout.ms";
+  private static final String RATE_LIMIT_TIMEOUT_MS_DOC =
+      "How much to wait for the necessary permits to proceed, in milliseconds. A request that "
+          + "fails to acquire the necessary permits within the timeout will fail with HTTP 429. "
+          + "Default is 0ms.";
+  private static final long RATE_LIMIT_TIMEOUT_MS_DEFAULT = 0;
+
+  public static final String RATE_LIMIT_DEFAULT_COST_CONFIG = "rate.limit.default.cost";
+  private static final String RATE_LIMIT_DEFAULT_COST_DOC =
+      "The default cost for resources/methods without an explicit cost configured via"
+          + "rate.limit.costs. Default is 1.";
+  private static final int RATE_LIMIT_DEFAULT_COST_DEFAULT = 1;
+
+  public static final String RATE_LIMIT_COSTS_CONFIG = "rate.limit.costs";
+  private static final String RATE_LIMIT_COSTS_DOC =
+      "Map of rate-limit cost per endpoint. The map should be in the form 'api1=10,api2=20'.";
```

Comment:
Can the doc give an explicit example here of what "api" looks like. eg is it "api.v3.produce.*" or "api.v3.produce" or "/v3/clusters/{clusterId}/topics/{topicName}/records" or...

Reply:
Done.